### PR TITLE
[ADD] enrich_werkzeug

### PR DESCRIPTION
**Pull request by Laurent-Corron**

### CODE REVIEW

File: test/data/testWerkzeug.log [hunk 1]
```diff
@@ -0,0 +1,33 @@
+2019-12-19 08:42:31,914 19646 INFO odoo13-dev odoo.addons.base.models.ir_http: Generating routing map for key None
+2019-12-19 08:42:32,117 19627 INFO ? odoo.addons.queue_job.jobrunner.runner: starting
+2019-12-19 08:42:32,118 19627 INFO ? odoo.addons.queue_job.jobrunner.runner: initializing database connections
+2019-12-19 08:42:32,135 19627 INFO ? odoo.addons.queue_job.jobrunner.runner: database connections ready
+2019-12-19 08:42:32,825 19646 INFO odoo13-dev werkzeug: 127.0.0.1 - - [19/Dec/2019 08:42:32] "GET /web HTTP/1.1" 200 - 474 0.351 0.565
+2019-12-19 08:42:32,892 19649 INFO odoo13-dev werkzeug: 127.0.0.1 - - [19/Dec/2019 08:42:32] "GET /web/static/lib/bootstrap/scss/web.assets_common/_functions.scss.css HTTP/1.1" 304 - 3 0.002 0.006
```
Comment (sbidoul) [inline test/data/testWerkzeug.log]: please trim this file a little bit so it's easier to read and compare results. If the goal is testing the werkzeug regexp, then one line is enough

File: src/loog/parser.py [hunk 2]
```diff
@@ -42,3 +42,29 @@ def parse_stream(stream: Iterable[str]) -> Iterator[Mapping[str, str]]:
...
+    r"^"
+    r"(?P<ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) - - "
+    r"\[\d{2}\/\w{3}\/\d{4} \d{2}\:\d{2}\:\d{2}\] "
+    r"\"(?P<method>\w+) "
+    r"(?P<url>\S*) \S* "
+    r"(?P<response>\d*) - "
+    r"(?P<sql_count>\d*) "
+    r"(?P<sql_time>\d*.\d{3}) "
+    r"(?P<python_time>\d*.\d{3})"
+    r"$"
```
Comment (sbidoul) [inline src/loog/parser.py]:
```suggestion
    r".*$"
```

File: src/loog/parser.py [hunk 3]
```diff
@@ -42,3 +42,29 @@ def parse_stream(stream: Iterable[str]) -> Iterator[Mapping[str, str]]:
...
+
+ODOO_werkzeug_RE = re.compile(
+    r"^"
+    r"(?P<ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) - - "
+    r"\[\d{2}\/\w{3}\/\d{4} \d{2}\:\d{2}\:\d{2}\] "
+    r"\"(?P<method>\w+) "
+    r"(?P<url>\S*) \S* "
+    r"(?P<response>\d*) - "
+    r"(?P<sql_count>\d*) "
+    r"(?P<sql_time>\d*.\d{3}) "
```
Comment (sbidoul) [inline src/loog/parser.py]:
```suggestion
    r"(?P<sql_time>\d*\.\d{3}) "
```

File: src/loog/parser.py [hunk 4]
```diff
@@ -42,3 +42,29 @@ def parse_stream(stream: Iterable[str]) -> Iterator[Mapping[str, str]]:
...
+    r"(?P<python_time>\d*.\d{3})"
```
Comment (sbidoul) [inline src/loog/parser.py]:
```suggestion
    r"(?P<python_time>\d*\.\d{3})"
```

**Comment by sbidoul** [comment]
Looks good, thanks.

I propose a slightly less strict regexp:

```
^(?P<ip>\S*) .*?] \"(?P<method>\S+) (?P<url>\S+) .*?\" (?P<response>\S+) - ((?P<sql_count>\S+) (?P<sql_time>\S+) (?P<python_time>\S+))?.*$
```

* The ip could be ipv6
* `.*` at the end in case the log is adapted with additional information (we sometimes do this)
* the last group (sql_count, sql_time, python_time) is made optional because it appeared in Odoo 11 or 12 only - can you add a test with and without it?

Also could you look in the nginx or apache logging documentation to find more precise and better names for fields ip (client_ip?), url (uri? path?), reponse (http status?).

**Comment by sbidoul** [comment]
Thinking about it it's probably better to put the enriched data in a sub dictionary. So a new key `werkzeug` with the new fields inside.

**Comment by Laurent-Corron** [comment]
> * the last group (sql_count, sql_time, python_time) is made optional because it appeared in Odoo 11 or 12 only - can you add a test with and without it?

it returns `None` in the key, not sure if there is an option to not create them in that case.

**Comment by sbidoul** [comment]
> option to not create them in that case

`{k: v for k, v in mo.groupdict().items() if v is not None}`

**Pull request merged**